src/core/util/http_client/httpcli.h

```diff
@@ -252,16 +255,23 @@ class HttpRequest : public InternallyRefCounted<HttpRequest> {
   RefCountedPtr<HandshakeManager> handshake_mgr_ ABSL_GUARDED_BY(mu_);
   bool cancelled_ ABSL_GUARDED_BY(mu_) = false;
   grpc_http_parser parser_ ABSL_GUARDED_BY(mu_);
-  std::vector<grpc_resolved_address> addresses_ ABSL_GUARDED_BY(mu_);
+  std::vector<grpc_event_engine::experimental::EventEngine::ResolvedAddress>
+      addresses_ ABSL_GUARDED_BY(mu_);
   size_t next_address_ ABSL_GUARDED_BY(mu_) = 0;
   int have_read_byte_ ABSL_GUARDED_BY(mu_) = 0;
   grpc_iomgr_object iomgr_obj_ ABSL_GUARDED_BY(mu_);
   grpc_slice_buffer incoming_ ABSL_GUARDED_BY(mu_);
   grpc_slice_buffer outgoing_ ABSL_GUARDED_BY(mu_);
   grpc_error_handle overall_error_ ABSL_GUARDED_BY(mu_) = absl::OkStatus();
+  // TODO(yijiem): remove these once event_engine_dns_non_client_channel
+  // experiment is fully enabled.
+  bool use_event_engine_dns_resolver_ = false;
```

Comment:
Note that this default is never used, since the variable is always set in the constructor.

Reply:
good catch.